Yoast/wordpress-seo
# Create endpoint to get an indexable

## Summary

This PR can be summarized in the following changelog entry:

* Introduce REST API endpoint for getting an indexable post or term.

## Relevant technical choices:

* Introduced interface `WPSEO_Endpoint_Storable` which isn't needed for this pull request, but will be used later.

## Test instructions

This PR can be tested by following these steps:

For testing the endpoint you have to use a nonce. This is a 'secret' string that should be send within your request. To get this nonce you can go to the WordPress admin dashboard and in your console you can do: `wpApiSettings.nonce`. If everything works wel you get the nonce.

To use the nonce you have to append this to the request url: `url?_wpnonce={the nonce}`. This means in case of the post indexable: `your-test-domain/wp-json/yoast/v1/indexables/post/{a-post-id}?_wpnonce={the nonce}`

* Sets all relevant meta values for a post and a term. Bring some variation in the values you store.
* Checkout this branch.
* In your browser open the endpoint by `your-test-domain/wp-json/yoast/v1/indexables/post/{a-post-id}` and open `your-test-domain/wp-json/yoast/v1/indexables/term/{a-term-id}`
* Verify if the result is matching the specs given in #9218 

## Quality assurance


Fixes #9218
